api/apps/v1alpha1/nemo_datastore_types.go

```diff
@@ -88,20 +105,49 @@ type NemoDatastoreStatus struct {
 	State             string             `json:"state,omitempty"`
 }
 
-type NemoDataStoreParams struct {
-	AppVersion    string `json:"appVersion"`
-	GiteaEndpoint string `json:"giteaEndpoint"`
-	GiteaSecret   string `json:"giteaSecret"`
-	DatabaseURL   string `json:"databaseURL"`
-	DatabaseHost  string `json:"databaseHost"`
-	DatabasePort  string `json:"databasePort"`
-	DBSecret      string `json:"dbSecret"`
+type ObjectStoreConfig struct { // e.g. Minio, s3
+	// ObjectStoreCredentials stores the configuration to retrieve the object store credentials
+	Credentials ObjectStoreCredentials `json:"credentials"`
+
+	// +kubebuilder:default:=true
+	ServeDirect bool `json:"serveDirect,omitempty"`
+
+	// endpoint is the fully qualidfied object store endpoint
+	Endpoint string `json:"endpoint"`
+	// BucketName is the bucket where LFS files will be stored
+	BucketName string `json:"bucketName"`
+	// Region is the region where bucket is hosted
+	Region string `json:"region"`
+	// SSL enable ssl for object store transport
+	SSL bool `json:"ssl"`
+}
+
+type ObjectStoreCredentials struct {
+	// User is the non-root username for a NEMO Service in the object store.
+	User string `json:"user"`
+
+	// SecretName is the name of the secret which has the object credentials for a NEMO service user.
+	SecretName string `json:"secretName"`
 
-	EnvConfigMap string `json:"envConfigmap"`
-	EnvSecret    string `json:"envSecret"`
+	// PasswordKey is the name of the key in the `CredentialsSecret` secret for the object store credentials.
+	PasswordKey string `json:"passwordKey"`
+}
 
-	InitContainerImage   string   `json:"initContainerImage,omitempty"`
-	InitContainerCommand []string `json:"initContainerCommand,omitempty"`
+type ExternalDatabase struct {
```

Comment:
Where is this object used?

Reply:
It's now replaced by `DatabaseConfig` in nemo_common_types.go`

Removing the struct as it's no longer used